### DISCUSSION TITLE
UseState vs UseLocalStorageState

### DISCUSSION THREAD

**Discussion by PeetThesing** [neuefische/questions#783] [Web] [answered]
Hello,

we implemented a favorite-feature for our app. Everything works fine when we use useState. We are aware that using useLocalStorageState would be a better option, but if we try it, we get an error complaining about our array-methods. The problem is that favorites is at first an array (like we wanted it to be) but then, when the data is fetched (after is loading), becomes a boolean.
Why?

Here is our [PR](https://github.com/PeetThesing/capstone-project_snippet-manager/pull/52)

We appreciate your suggestions.

**Comment by doemser** [comment, marked as answer]
Can you please provide more information like screenshots and error messages?

Problems with localStorage usage often occur because of an "old" localStorage state, that then collides with a newer approach, so make sure, that you clear your localStorage once in the inspector, so we can exclude this as cause of your error.

**Reply by PeetThesing** [reply]
That was the error: ![Bildschirmfoto 2023-11-27 um 11 58 14](https://github.com/neuefische/questions/assets/87306048/f09a7622-0288-4bc0-aaff-1952c47df0d5)
But clearing  the local storage state helped, but why? Why changes the Array to a boolean when the storage isn't cleared?
